WordPressAuthenticator/Model/LoginFields.swift

```diff
@@ -56,11 +56,50 @@ public class LoginFields: NSObject {
 
         return loginFields
     }
+
+    /// Using a convenience initializer for its Objective-C usage in unit tests.
+    convenience init(username: String,
+                     password: String,
+                     siteAddress: String,
+                     multifactorCode: String,
+                     nonceInfo: SocialLogin2FANonceInfo?,
+                     nonceUserID: Int,
+                     restrictToWPCom: Bool,
+                     emailAddress: String,
+                     meta: LoginFieldsMeta,
+                     storedCredentials: SafariStoredCredentials?) {
+        self.init()
+        self.username = username
+        self.password = password
+        self.siteAddress = siteAddress
+        self.multifactorCode = multifactorCode
+        self.nonceInfo = nonceInfo
+        self.nonceUserID = nonceUserID
+        self.restrictToWPCom = restrictToWPCom
+        self.emailAddress = emailAddress
+        self.meta = meta
+        self.storedCredentials = storedCredentials
+    }
+}
+
+extension LoginFields {
+    func copy() -> LoginFields {
+        .init(username: username,
+              password: password,
+              siteAddress: siteAddress,
+              multifactorCode: multifactorCode,
+              nonceInfo: nonceInfo,
+              nonceUserID: nonceUserID,
+              restrictToWPCom: restrictToWPCom,
+              emailAddress: emailAddress,
+              meta: meta.copy(),
+              storedCredentials: storedCredentials)
+    }
 }
 
 /// A helper class for storing safari saved password information.
 ///
-class SafariStoredCredentials {
+struct SafariStoredCredentials {
```

Comment:
Dumb question: why does this need to be a `struct`? And should we update the comment since it's no longer a class?

Reply:
Great question! Originally, I thought it's better to use `Struct` so that a new struct is created when copying a `LoginFields` instead of copying by reference. However, I decided to revert this change since I don't understand the existing usage of `SafariStoredCredentials` enough. Reverted in https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/668/commits/0a3bf419f07ebbd623e92ed9727211d549b611ea